docs/content/src/learn/tutorials/messaging-integrations/messaging-with-kafka/README.md

```diff
@@ -0,0 +1,138 @@
+# Messaging with Kakfa
+
+Kafka is a distributed, partitioned, replicated commit log service. It provides the functionality of a publish-subscribe messaging system with a unique design. Kafka mainly operates based on a topic model. A topic is a category or feed name to which records get published. Topics in Kafka are always multi-subscriber.
+
+This guide walks you through the process of messaging with Apache Kafka using Ballerina language. 
+
+## What you'll build
+
+To understand how you can use Kafka for 'publish-subscribe' messaging, lets consider a real-world use case of a Product Management System. This product management system consists of a product admin portal where the product administrator can update the price for a product. Based on the content of the message sent to the product admin service, it is filtered to different partitions in the Kafka Topic.  
+
+The following figure illustrates the publish-subscribe messaging use-case. 
+
+![messaging_with_kafka](../../../../assets/img/messaging_with_kafka.png "Messaging with Kafka")
+
+<!-- INCLUDE_MD: ../../../../tutorial-prerequisites.md -->
+
+<!-- INCLUDE_MD: ../../../../tutorial-get-the-code.md -->
+
+## Implementation
+
+####1. Create a new Ballerina project named `messaging-with-kafka`.
+
+```bash
+$ ballerina new messaging-with-kafka
+```
+
+####2. Navigate to the messaging-with-kafka directory and add a new module named `product_management_system` to the project.
+
+```bash
+$ ballerina add product_management_system
+```
+
+####3. Open the project with VS Code. The project structure will be similar to the following.
```

Comment:
This displays as a heading so it's better to shorten it and have a description underneath.

Reply:
Shortened heading